home/models.py

```diff
@@ -1830,30 +1852,29 @@ def clean(self):
                 ValidationError("All WhatsApp templates need a name.")
             )
 
-        message = self.message
-
-        # TODO: Explain what this does, or find a cleaner implementation
-
-        # Checks for mismatches in the number of opening and closing brackets.  First from right to left, then from left to right
-        # TODO: Currently "{1}" is allowed to pass and throws an error.  Add a check for this, or redo this section in a cleaner way
-        right_mismatch = re.findall(r"(?<!\{){[^{}]*}\}", message)
-        left_mismatch = re.findall(r"\{{[^{}]*}(?!\})", message)
-        mismatches = right_mismatch + left_mismatch
-
         example_values = self.example_values.raw_data
         for ev in example_values:
             if "," in ev["value"]:
                 errors["example_values"] = ValidationError(
                     "Example values cannot contain commas"
                 )
 
-        if mismatches:
+        message = self.message
+        # Matches "{1}" and "{11}", not "{{1}", "{a}" or "{1 "
+        single_braces = re.findall(r"[^{]{(\d*?)}", message)
+
+        if single_braces:
             errors.setdefault("message", []).append(
                 ValidationError(
-                    f"Please provide variables with matching braces. You provided {mismatches}."
+                    f"Please provide variables with valid double braces. You provided single braces {single_braces}."
                 )
             )
 
+        brace_mismatches = self.check_matching_braces(message)
+
+        if brace_mismatches:
```

Comment:
These if statements might be a good candidate for the new walrus operator, e.g. `if brace_mismatches := self.check_matching_braces(message):`

Reply:
Have added TODO to replace with PyParsing as discussed